homework02/allan/string.h

```diff
@@ -0,0 +1,41 @@
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+
+class String
+{
+public:
+    String();
+    String(const char *);
+    String(const String& other);
+                String(const char *, int capactiy);
```

Comment:
It is a bit odd as API to allow the user to pass the capacity here... I would have more expected the size? But why not, after all, its your string, not mine!